core/src/main/java/org/mule/runtime/core/internal/processor/strategy/ComponentProcessor.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.internal.processor.strategy;
+
+import org.mule.runtime.core.api.processor.ReactiveProcessor;
+
+/**
+ * Specialization of {@link ReactiveProcessor} that allows to perform certain optimizations when the processor is not blocking, as
```

Comment:
allows to perform -> allows performing

Reply:
I've used the original motivation for the introduction of this inner processor to write the jdoc (originally named OperationInnerProcessor MULE-17939). I've changed the name to ComponentInnerProcessor. The idea is to apply the processing strategy only to the operation when there is a policy involved.